Prusa-CW-Firmware/main.cpp

```diff
@@ -534,11 +560,13 @@ void PID() {
   if ((summErr > 10000) || (summErr < -10000))
     summErr = 10000;
 
-  diffTemp = actualTemp - oldSpeed;
+  diffTemp = actualTemp;	// "- oldSpeed;" which was always 0
+
   newSpeed = P * errValue + I * summErr + D * diffTemp;
   if (newSpeed > 100) {
-    newSpeed = 100;
+    return 100;
```

Comment:
Only single return point is recomended for non-trivial functions. (It is easier to maintain, and debug.)